### PR TITLE
Allow empty indexed arrays.

### LINKED ISSUE #3152: ValueError: For an array_value, subvalues must either all be indexed or all excluded from indexes.
Getting `ValueError: For an array_value, subvalues must either all be indexed or all excluded from indexes.` error when querying entities with empty arrays.

Edit: I tried getting around this with projection queries where possible but I need to modify the entities after fetching them, which you can't do with the results of a projection query. So this is a pretty big issue as there is no way around it.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Fixes #3152.

### CODE REVIEW

File: datastore/tests/unit/test_helpers.py [hunk 1]
```diff
@@ -135,6 +135,27 @@ def test_mismatched_value_indexed(self):
         with self.assertRaises(ValueError):
             self._call_fut(entity_pb)
 
+    def test_index_mismatch_ignores_empty_list(self):
+        from google.cloud.proto.datastore.v1 import entity_pb2
+        from google.cloud.datastore.helpers import _new_value_pb
+
+        _PROJECT = 'PROJECT'
+        _KIND = 'KIND'
+        _ID = 1234
```
Comment (dhermes) [inline datastore/tests/unit/test_helpers.py]: Just for fun: can you make this "normal" names for locals, i.e. `project`, `kind` and `id_`?

File: datastore/tests/unit/test_helpers.py [hunk 2]
```diff
@@ -135,6 +135,27 @@ def test_mismatched_value_indexed(self):
...
+        entity_pb = entity_pb2.Entity()
+        entity_pb.key.partition_id.project_id = _PROJECT
+        entity_pb.key.path.add(kind=_KIND, id=_ID)
+
+        array_val_pb = _new_value_pb(entity_pb, 'baz')
+        array_pb = array_val_pb.array_value.values
+
+        # unindexed_value_pb1 = array_pb.add()
+        # unindexed_value_pb1.integer_value = 10
```
Comment (dhermes) [inline datastore/tests/unit/test_helpers.py]: Please remove commented out code.

File: datastore/tests/unit/test_helpers.py [hunk 3]
```diff
@@ -135,6 +135,27 @@ def test_mismatched_value_indexed(self):
...
+
+        entity = self._call_fut(entity_pb)
+        entity_dict = dict(entity)
+        self.assertIsInstance(entity_dict['baz'], list)
```
Comment (dhermes) [inline datastore/tests/unit/test_helpers.py]: Why not just check equality for what the value is?

File: datastore/tests/unit/test_helpers.py (hunk 1 above)
Comment (lukesneeringer) [inline datastore/tests/unit/test_helpers.py]: Every single other test in this module uses those attribute names.

**Comment by lukesneeringer** [comment]
Closing and re-opening.

**Pull request closed without merging**